File: papara-sdk/src/main/java/com/mobillium/paparasdk/utils/PaparaLogger.java
package com.mobillium.paparasdk.utils;

import android.util.Log;

import com.mobillium.paparasdk.Papara;

/**
 * Created by oguzhandongul on 18/10/2016.
 */

public class PaparaLogger {
    private static String TAG = "PAPARA_SDK";
    private static String TAG_PARAMS = "PAPARA_SDK_PARAMS";

    public PaparaLogger() {
    }


    public static void writeErrorLog(String message) {
        if (Papara.getInstance().isDebugEnabled()) {
            Log.e(TAG, message);
        }
    }

    public static void writeInfoLog(String message) {
        if (Papara.getInstance().isDebugEnabled()) {
            Log.i(TAG, message);
        }
    }

    public static void writeResponseLog(String message) {
        if (Papara.getInstance().isDebugEnabled()) {
            Log.i(TAG_PARAMS, message);
        }
    }

    public static void writeDebugLog(String message) {
        if (Papara.getInstance().isDebugEnabled()) {
            Log.d(TAG, message);
        }
    }
}

File: papara-sdk/src/main/java/com/papara/sdk/utils/PaparaLogger.kt
package com.papara.sdk.utils

import android.util.Log
import com.papara.sdk.Papara

object PaparaLogger {

    private const val TAG = "PAPARA_SDK"

    @JvmStatic
    fun writeErrorLog(message: String?) {
        if (Papara.getInstance().isDebugEnabled) {
            Log.e(TAG, message.orEmpty())
        }
    }

    @JvmStatic
    fun writeInfoLog(message: String?) {
        if (Papara.getInstance().isDebugEnabled) {
            Log.i(TAG, message.orEmpty())
        }
    }

    @JvmStatic
    fun writeResponseLog(message: String?) {
        if (Papara.getInstance().isDebugEnabled) {
            Log.i(TAG, message.orEmpty())
        }
    }

    @JvmStatic
    fun writeDebugLog(message: String?) {
        if (Papara.getInstance().isDebugEnabled) {
            Log.d(TAG, message.orEmpty())
        }
    }

    @JvmStatic
    fun writeSdkErrorLog() {
        if (Papara.getInstance().isDebugEnabled) {
            Log.e(TAG, "You must initialize the Papara SDK first")
        }
    }
}